status/status.go

```diff
@@ -251,26 +251,31 @@ func (s *Status) ReportSnapshotStatus(ctx context.Context, reporter ReporterInte
 	for _, integrationTestStatusDetail := range integrationTestStatusDetails {
 		if srs.IsNewer(integrationTestStatusDetail.ScenarioName, integrationTestStatusDetail.LastUpdateTime) {
 			s.logger.Info("Integration Test contains new status updates", "scenario.Name", integrationTestStatusDetail.ScenarioName)
+			srs.SetLastUpdateTime(integrationTestStatusDetail.ScenarioName, integrationTestStatusDetail.LastUpdateTime)
```

Comment:
Why are you setting new time here?

Reply:
Because if we encounter an error while calling `s.generateTestReport()` or `reporter.ReportStatus`, we'd be reporting old status to the Snapshot's `git-reporter-status` annotation.

Reply:
with this, if ReportStatus fails, we will never retry

Reply:
because `if srs.IsNewer` will be false

Reply:
Yes! I missed that. Thanks!
I've reverted this change. PTAL